src/main/java/loqor/ait/core/tardis/dim/TardisDimension.java

```diff
@@ -70,6 +73,18 @@ public static Optional<Tardis> get(World world) {
 
         return Optional.ofNullable(TardisManager.with(world, ((o, manager) -> manager.demandTardis(o, uuid))));
     }
+    public static void withTardis(ServerWorld world, Consumer<ServerTardis> consumer) {
```

Comment:
ow.

In theory, there's a better implementation... We could use our own `ServerWorld` implementation in multidim, so we could use a `instanceof` check instead of doing all this. Besides, with our own implementation we could just give it a tardis instance to cache. Same goes for `ServerTardis`.

Reply:
thats a good idea but itd be something id do on another PR
ill do a PR for `TardisDimension` refactoring and performance enhancing